/**
 * Copyright 2020 Huawei Technologies Co., Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
syntax = "proto3";
package mindspore;
import "attr.proto";
import "tensor.proto";

message DynamicIdx {
  int32 idx = 1;
  int32 num = 2;
}

message NodeDef {
  string op = 2;
  map<string, AttrValue> attrs = 3;
  repeated Tensor inputs = 4;
  repeated Tensor outputs = 5;
  map<string, DynamicIdx> dym_inputs = 6;
  map<string, DynamicIdx> dym_outputs = 7;
}
